sector-builder/src/scheduler.rs

```diff
@@ -306,6 +316,52 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
         self.checkpoint().expects(FATAL_SNPSHT);
     }
 
+    /// Check the validity of the all sealed sectors.
+    pub fn get_all_sealed_sector_status(&self) -> Result<Vec<SectorStatus>> {
+        use rayon::prelude::*;
+        self.state
+            .sealed
+            .sectors
+            .par_iter()
+            .map(|(id, _)| self.get_sector_status(*id))
+            .collect()
+    }
+
+    /// Check the validity of the given sector.
+    pub fn get_sector_status(&self, id: SectorId) -> Result<SectorStatus> {
+        if let Some(sector) = self.state.sealed.sectors.get(&id) {
+            let expected_checksum = &sector.checksum[..];
+            let expected_len = sector.len;
+
+            let sector_path = self
+                .sector_store
+                .manager()
+                .sealed_sector_path(&sector.sector_access);
+
+            // Check length
+            let m = std::fs::metadata(&sector_path)?;
+            if m.len() != expected_len {
```

Comment:
I don't understand why this check is useful. The checksum is sufficient proof that the sector is in a good state. 

If we don't perform this check, we can drop the `len` field from `SealedSectorMetadata`.

Reply:
Because it is much cheaper to check this, and can already detect a bunch of other issues. checksum is really only needed when we have the exact same byte length